app/graphql/mutations/create_event_mutation.rb

```diff
@@ -0,0 +1,36 @@
+require_relative 'mutation_root'
+require_relative 'transformers/panoptes_classification'
+
+module Mutations
+  class CreateEventMutation < Mutations::BaseMutation
+    null true
+
+    argument :event_payload, [String], required: true, description: 'list of events in json format'
+
+    field :errors, [String], null: false
+
+    def resolve(event_payload:)
+      # TODO: Add guard here to only ingest data that meets the schema
+      # avoid ouroboros data and panoptes talk data until we have verified
+      # the schema conformance here
+      # return unless model.type && model.source == panoptes classificaiton
+      
+      # TODO: authorisation here
+      if event_payload.nil?
+        return {errors: [{"message" => "ArgumentError"}]}
+      end
+      
+      events_list = []
+      event_payload.each do |event|
+        event_hash = eval(event)
+        prepared_payload = Transformers::PanoptesClassification.new(event_hash).transform
+        events_list.append(Event.new(prepared_payload))
+      end
+
+      events_list.each do |event|
+        return { errors: event.errors.full_messages } unless event.save
```

Comment:
If one event fails to save, this will abort halfway through the event list, but not roll back (because it's not wrapped inside a transaction). Is that the behaviour you wanted?